Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -101,65 +102,246 @@ public final class ActorQueue<ActorType: Actor>: @unchecked Sendable {
         // Crashing here means that this queue is being sent tasks either before an execution context has been set, or
         // after the execution context has deallocated. An ActorQueue's execution context should be set in the adopted
         // actor's `init` method, and the ActorQueue should not exceed the lifecycle of the adopted actor.
         weakExecutionContext!
     }
+
+    fileprivate struct ActorTask: Sendable {
+        init(executionContext: ActorType, task: @escaping @Sendable (isolated ActorType) async -> Void) {
+            self.executionContext = executionContext
+            self.task = task
+        }
+        
+        let executionContext: ActorType
+        let sempahore = Semaphore()
+        let task: @Sendable (isolated ActorType) async -> Void
+    }
+
+    // MARK: Private
+
     /// The actor on whose isolated context our tasks run.
     /// We must use`weak` here to avoid creating a retain cycle between the adopted actor and this actor queue.
     ///
     /// We will assume this execution context always exists for the lifecycle of the queue because:
     /// 1. The lifecycle of any `ActorQueue` must not exceed the lifecycle of its adopted `actor`.
     /// 2. The adopted `actor` must set itself as the execution context for this queue within its `init` method.
     private weak var weakExecutionContext: ActorType?
+}
 
-    private struct ActorTask {
-        let executionContext: ActorType
-        let task: @Sendable (isolated ActorType) async -> Void
+extension Task {
+    /// Runs the given nonthrowing operation asynchronously
+    /// as part of a new top-level task on behalf of the current actor.
+    /// The operation will not execute until all prior tasks have
+    /// completed or suspended.
+    ///
+    /// Use this function when creating asynchronous work
+    /// that operates on behalf of the synchronous function that calls it.
+    /// Like `Task.detached(priority:operation:)`,
+    /// this function creates a separate, top-level task.
+    /// Unlike `Task.detached(priority:operation:)`,
+    /// the task created by `Task.init(priority:operation:)`
+    /// inherits the priority and actor context of the caller,
+    /// so the operation is treated more like an asynchronous extension
+    /// to the synchronous operation.
+    ///
+    /// You need to keep a reference to the task
+    /// if you want to cancel it by calling the `Task.cancel()` method.
+    /// Discarding your reference to a detached task
+    /// doesn't implicitly cancel that task,
+    /// it only makes it impossible for you to explicitly cancel the task.
+    ///
+    /// - Parameters:
+    ///   - priority: The priority of the task.
+    ///     Pass `nil` to use the priority from `Task.currentPriority`.
+    ///   - actorQueue: The queue on which to enqueue the task.
+    ///   - operation: The operation to perform.
+    @discardableResult
+    public init<ActorType: Actor>(
+        priority: TaskPriority? = nil,
```

Comment:
should remove... priority is the same as queue priority

Reply:
now we're respecting it! removed from `FIFOQueue` though where it makes less sense